### DISCUSSION TITLE
How to exclude the header row(s) from the search?

### DISCUSSION THREAD

**Discussion by datatraveller1** [BurntSushi/ripgrep#1950] [General] [answered]
I want to search only in the data rows (beginning with line number 2) of a CSV file excluding the first header row.
Is it possible? If not, may it be possible to add an option skip-header=n where n means the number of skipped header rows?

**Comment by BurntSushi** [comment, marked as answer]
ripgrep isn't a tool for parsing CSV, which would be a hard requirement for adding your proposed flag. (A single CSV record can span multiple lines.)

I would recommend using `tail` if you know your header row is one line. Otherwise, use xsv's slice sub-command and pipe the output to ripgrep.

**Reply by datatraveller1** [reply]
Yes, ripgrep isn't a tool for parsing CSV. Therefore, there is no need to care about CSV rules (RFC 4180).
To be more precise, I most often use tab separated files with no tabs in a field and no newlines in a field. For these files no CSV parsing is needed. (BTW, personally I would never use multiline fields in a CSV header).
That said, an even better option would be `--header-lines=n`
Then, ripgrep should always print the first n header lines and would not use these lines for searching.
Does this suggestion make more sense?

**Reply by BurntSushi** [reply]
As I said, I think `tail` is a better fit.

**Reply by datatraveller1** [reply]
Thank you. Do you suggest to pipe the result of `tail +2 file.txt` to ripgrep?
e.g. `tail +2 file.txt | rg "search_string"`
It works but is not fast for very big files (at least on Windows). Further, there has to be considered that `--line-number` prints one number less according to the original file (this can even be wished but one has to bear in mind).
So no problem, I can use it and I can print the header first with with head -1:
`head -1 file.txt >out.txt`
 `tail +2 file.txt | rg "search_string" >>out.txt`
Much appreciated. Thank you for the great ripgrep tool!
